tests/e2e/nvme_metrics.go

```diff
@@ -0,0 +1,166 @@
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/framework/kubectl"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+var _ = Describe("[ebs-csi-e2e] [single-az] NVMe Metrics", func() {
+	f := framework.NewDefaultFramework("ebs")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+
+	var (
+		cs        clientset.Interface
+		ns        *v1.Namespace
+		ebsDriver driver.PVTestDriver
+	)
+
+	BeforeEach(func() {
+		cs = f.ClientSet
+		ns = f.Namespace
+		ebsDriver = driver.InitEbsCSIDriver()
+	})
+
+	It("should create a stateful pod and read NVMe metrics from node plugin", func() {
+		pod := testsuites.PodDetails{
+			Cmd: "while true; do echo $(date -u) >> /mnt/test-1/out.txt; sleep 5; done",
+			Volumes: []testsuites.VolumeDetails{
+				{
+					CreateVolumeParameters: map[string]string{
+						ebscsidriver.VolumeTypeKey: awscloud.VolumeTypeGP3,
+						ebscsidriver.FSTypeKey:     ebscsidriver.FSTypeExt4,
+					},
+					ClaimSize: driver.MinimumSizeForVolumeType(awscloud.VolumeTypeGP3),
+					VolumeMount: testsuites.VolumeMountDetails{
+						NameGenerate:      "test-volume-",
+						MountPathGenerate: "/mnt/test-",
+					},
+				},
+			},
+		}
+
+		By("Setting up pod and volumes")
+		tpod, _ := pod.SetupWithDynamicVolumes(cs, ns, ebsDriver)
+		defer tpod.Cleanup()
+
+		By("Creating the stateful pod")
+		tpod.Create()
+		tpod.WaitForRunning()
+
+		By("Getting name of the node where test pod is running")
+		p, err := cs.CoreV1().Pods(ns.Name).Get(context.TODO(), tpod.GetName(), metav1.GetOptions{})
+		framework.ExpectNoError(err, "Failed to get pod")
+		nodeName := p.Spec.NodeName
+		Expect(nodeName).NotTo(BeEmpty(), "Node name should not be empty")
+
+		By("Finding the CSI node plugin on the node where test pod is running")
+		csiPods, err := cs.CoreV1().Pods("kube-system").List(context.TODO(), metav1.ListOptions{
+			LabelSelector: "app=ebs-csi-node",
+		})
+		framework.ExpectNoError(err, "Failed to get CSI node pods")
+
+		var csiNodePod *v1.Pod
+		for _, pod := range csiPods.Items {
+			if pod.Spec.NodeName == nodeName {
+				csiNodePod = &pod
+				break
+			}
+		}
+		Expect(csiNodePod).NotTo(BeNil(), "No CSI node pod found")
+
+		By("Setting up port-forwarding")
+		args := []string{"port-forward",
+			"-n", "kube-system",
+			csiNodePod.Name,
+			"3302:3302"}
+
+		go kubectl.RunKubectlOrDie("kube-system", args...)
+
+		By("Getting metrics from endpoint")
+		ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+		defer cancel()
+
+		metricsOutput, err := getMetricsWithRetry(ctx)
+		framework.ExpectNoError(err, "Failed to get metrics after retries")
+
+		By("Verifying NVMe metrics")
+		expectedMetrics := []string{
+			"total_read_ops",
+			"total_write_ops",
+			"total_read_bytes",
+			"total_write_bytes",
+			"read_io_latency_histogram",
+			"write_io_latency_histogram",
+		}
```

Comment:
Why less metrics here than in the unit test? Are we not expecting to also see metrics like `volume_queue_length` and `read_io_latency_histogram`?

Reply:
Included the full list in latest rev.